index.bs

```diff
@@ -681,14 +706,551 @@ id=] must be the same as the [=window handle=].
+
+            1. [=Set=](|serialized|, "attributes", |attributes|, false)
+
+        1. Let |remote value| be an [=object=] matching the <code>NodeValue</code>
+           production in the [=local end definition=], with the <code>objectId</code>
+           property set to the [=object id for an object=] |value|, and <code>value</code>
+           set to |serialized|, if |serialized| is not null.
+
+      <dt>|value| is a [=platform object=] that implements [=WindowProxy=]
+      <dd>1. Let |remote value| be an [=object=] matching the <code>WindowProxyValue</code>
+             production in the [=local end definition=], with the <code>objectId</code>
+             property set to the [=object id for an object=] |value|.
+
+      <dt>|value| is a [=platform object=]
+      <dd>1. Let |remote value| be an [=object=] matching the <code>ObjectValue</code>
+             production in the [=local end definition=], with the <code>objectId</code>
+             property set to the [=object id for an object=] |value|.
+
+      <dt>[=IsCallable=](|value|)
+      <dd>Let |remote value| be an [=object=] matching the <code>FunctionValue</code>
+          production in the [=local end definition=], with the <code>objectId</code>
+          property set to the [=object id for an object=] |value|.
+
+      <dt>Otherwise:
+      <dd>
+      1. let |serialized| be null.
+
+      1. If |value| is not in the |set of known objects|, and |max depth|
+         is not null and greater than 0, run the following steps:
+         1. Append |value| to the |set of known objects|
+
+         1. Let |serialized| be the result of [=serialize as a mapping=] given
+            [=EnumerableOwnPropertyNames=](|value|, key+value), |max depth|, |node
+            details| and |set of known objects|
+
+      1. Let |remote value| be an [=object=] matching the <code>ObjectValue</code> production
+         in the [=local end definition=], with the <code>objectId</code> property set
+         to the [=object id for an object=] |value|, and the <code>value</code> field
+         set to |serialized|.
+    </dl>
+
+  1. Return |remote value|
+
+Issue: the way we check types is all wrong
+
+Issue: Does it make sense to use the same depth parameter for nodes and objects
+in general?
+
+</div>
+
+<div algorithm>
+Issue: this assumes for-in works on iterators
+
+To <dfn>serialize as a list</dfn> given |iterable|, |max depth|,
+|node details| and |set of known objects|:
+
+  1. Let |serialized| be a new list.
+
+  1. For each |child value| in |iterable|:
+
+    1. Let |serialized child| be the result of [=serialize as a
+       remote value=] with arguments |child value|, |max
+       depth| - 1, |node details| and |set of known objects|.
+
+    1. Append |serialized child| to |serialized|.
+
+  1. Return [=CreateArrayFromList=](|serialized|)
+</div>
+
+<div algorithm>
+To <dfn>serialize as a mapping</dfn> given |iterable|, |max depth|,
```

Comment:
This is used to create `ObjectValue`s, and it looks like it creates something on the same form as `Object.entries`, an array of 2-tuples of key+value pairs. What I don't quite get is why, since they keys could only be strings. Is it to preserve iteration order for languages which have unordered dicts, like Python in the past?

Could we just use CBOR objects instead?

Reply:
It also allows it to work with non-string keys, which are important for maps and synbol-keyed objects.

Reply:
Hmm, some bits appear to be missing here then. This "serialize as a mapping" algorithm is only called when making an `ObjectValue` production, not any maps AFAICT. For objects, [EnumerableOwnPropertyNames](https://tc39.es/ecma262/#sec-enumerableownpropertynames) has a "If Type(key) is String" condition which means we could only get string keys here.

For plain objects, I really have to question if it makes sense to complicate the representation in order to preserve more information than you would get from `JSON.stringify` or structured clone as in `postMessage`. Neither preserves symbols.

Reply:
Let's file a follow-up issue about this.